kernel/libs/xarray/src/test.rs

```diff
@@ -0,0 +1,333 @@
+
+    assert!(cursor.set_mark(XMark::Mark1).is_err());
+    assert!(!cursor.is_marked(XMark::Mark1));
+}
+
+#[ktest]
+fn box_operate() {
+    let xarray_box: XArray<Box<i32>> = XArray::new();
+    let mut locked_xarray = xarray_box.lock();
+    let mut cursor_mut = locked_xarray.cursor_mut(0);
+    for i in 0..n!(100) {
+        if i % 2 == 0 {
+            cursor_mut.store(Box::new(i * 2));
+        }
+        cursor_mut.next();
+    }
+
+    cursor_mut.reset_to(0);
+    for i in 0..n!(100) {
+        if i % 2 == 0 {
+            assert_eq!(*cursor_mut.load().unwrap().as_ref(), i * 2);
+        } else {
+            assert!(cursor_mut.load().is_none());
+        }
+        cursor_mut.next();
+    }
+
+    let mut cursor = locked_xarray.cursor(0);
+    for i in 0..n!(100) {
+        if i % 2 == 0 {
+            assert_eq!(*cursor.load().unwrap().as_ref(), i * 2);
+        } else {
+            assert!(cursor.load().is_none());
+        }
+        cursor.next();
+    }
+}
+
+#[ktest]
+fn range() {
+    let xarray_arc: XArray<Arc<i32>> = XArray::new();
+    for i in 0..n!(100) {
+        let value = Arc::new(i * 2);
+        xarray_arc.lock().store((i * 2) as u64, value);
+    }
+
+    let mut count = 0;
+    let guard = disable_preempt();
+    for (index, item) in xarray_arc.range(&guard, n!(10)..n!(20)) {
+        assert_eq!(*item.as_ref() as u64, index);
+        count += 1;
+    }
+    assert_eq!(count, n!(5));
+}
+
+#[ktest]
+fn load_after_clear() {
+    let xarray_arc: XArray<Arc<i32>> = XArray::new();
+    init_continuous_with_arc(&xarray_arc, n!(100));
+
+    let guard = disable_preempt();
+    let mut cursor = xarray_arc.cursor(&guard, 100);
+    let mut locked_xarray = xarray_arc.lock();
+
+    let value = cursor.load().unwrap();
+    assert_eq!(*value.as_ref(), 100);
+
+    // Read the old data.
+    locked_xarray.clear();
+    let value = cursor.load().unwrap();
+    assert_eq!(*value.as_ref(), 100);
+
+    // Read the new data.
+    cursor.reset();
+    assert!(cursor.load().is_none());
+}
+
+static TEST_LEAKAGE: AtomicBool = AtomicBool::new(false);
+static DROP_COUNT: AtomicUsize = AtomicUsize::new(0);
+
+impl<P: NonNullPtr + Send + Sync> Drop for node::XNode<P> {
+    fn drop(&mut self) {
+        if TEST_LEAKAGE.load(Ordering::Acquire) {
+            DROP_COUNT.fetch_add(1, Ordering::Release);
+        }
+    }
+}
+
+#[ktest]
+fn leakage() {
+    fn finish_grace_period() {
+        let task = || {};
+        let _ = ostd::task::TaskOptions::new(task).data(()).spawn();
+        Task::yield_now();
+    }
+
+    // Drop the nodes created by the previous tests.
+    finish_grace_period();
+    TEST_LEAKAGE.store(true, Ordering::Release);
+    let xarray_arc: XArray<Arc<i32>> = XArray::new();
+    init_sparse_with_arc(&xarray_arc, (SLOT_SIZE * SLOT_SIZE / 2 + 1) as i32);
+    drop(xarray_arc);
+    // Drop the nodes created in the test.
+    finish_grace_period();
+
+    let count = DROP_COUNT.load(Ordering::Acquire);
```

Comment:
Nothing to synchronize here. Can we convert all `Release` and `Acquire` to `Relaxed`?